### PR TITLE
Utilities: Add the pcfconv utility

### PR DESCRIPTION
**Pull request by janso3**
This tool is able to convert an X11 PCF font to Serenity's font format. These pcf files are a fairly common bitmap font format, and with this we are easily able to use many of them.

pcfconv

### CODE REVIEW

**Comment by janso3** [comment]
Could also turn this into a parser in LibGfx, not sure what makes more sense.

File: Userland/Utilities/pcfconv.cpp [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+            TRY(m_stream.read_some(m_bitmap_data.data));
+            break;
+        case PCF_BDF_ENCODINGS: {
+            m_encoding.min_char_or_byte2 = TRY(read<i16>(format));
+            m_encoding.max_char_or_byte2 = TRY(read<i16>(format));
+            m_encoding.min_byte1 = TRY(read<i16>(format));
+            m_encoding.max_byte1 = TRY(read<i16>(format));
+            m_encoding.default_char = TRY(read<i16>(format));
+
+            size_t num = (m_encoding.max_char_or_byte2 - m_encoding.min_char_or_byte2 + 1)
```
Comment (nico) [inline Userland/Utilities/pcfconv.cpp:580]: Should we check that min < max?

File: Userland/Utilities/pcfconv.cpp [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+        u8 lo = code_point & 0xff;
+        table_index = (hi - m_encoding.min_byte1)
+                * (m_encoding.max_char_or_byte2 - m_encoding.min_char_or_byte2 + 1)
+            + lo - m_encoding.min_char_or_byte2;
+    }
+
+    if (table_index < 0)
+        return {};
+
+    if (table_index >= static_cast<ssize_t>(m_encoding.indices.size()))
```
Comment (nico) [inline Userland/Utilities/pcfconv.cpp:321]: Does this happen? Should we error out instead?

File: Userland/Utilities/pcfconv.cpp [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+        file = TRY(Core::File::standard_input());
+    } else {
+        file = TRY(Core::File::open(input_path, Core::File::OpenMode::Read));
+    }
+
+    auto buffer = TRY(file->read_until_eof());
+    auto pcf = TRY(PCFFile::create(buffer));
+
+    if (pcf->glyph_size().width() > 32 || pcf->glyph_size().height() > 32) {
+        outln(stderr, "At this time, glyphs may only be 32px wide");
```
Comment (nico) [inline Userland/Utilities/pcfconv.cpp:192]: "at most"

File: Userland/Utilities/pcfconv.cpp [hunk 4]
```diff
@@ -0,0 +1,643 @@
...
+public:
+    static ErrorOr<NonnullRefPtr<PCFFile>> create(ReadonlyBytes);
+
+    ErrorOr<String> construct_filename() const;
+
+    Optional<i16> glyph_index_for(u16 code_point) const;
+    ErrorOr<void> draw_glyph(u16 index, Gfx::GlyphBitmap&) const;
+    u8 glyph_width(u16 index) const { return m_glyphs.at(index).width; }
+    u8 baseline() const;
+    size_t highest_codepoint() const;
```
Comment (nico) [inline Userland/Utilities/pcfconv.cpp:49]: Can this be larger than u16? The mapping only has 16bit values for min and max afaict.

**Review by nico — COMMENTED** [review]
Not sure where this belongs either! I suppose we can start here.

Given its X11 heritage, we might want to keep it at arm's length, and promote high quality bundled serenity bitmap fonts -- so maybe this should not move to LibGfx. But not sure.

**Pull request closed without merging**